shpc/main/registry/__init__.py

```diff
@@ -128,19 +135,17 @@ def _sync(
         local=None,
         sync_registry=None,
     ):
-        # Registry to sync from
-        sync_registry = sync_registry or self.settings.sync_registry
-
         # Create a remote registry with settings preference
-        Remote = GitHub if "github.com" in sync_registry else GitLab
-        remote = Remote(sync_registry, tag=tag)
-        local = self.get_registry(local or self.settings.filesystem_registry)
+        remote = self.get_registry(
+            sync_registry or self.settings.sync_registry, tag=tag
+        )
+        local = self.get_registry(local) if local else self.filesystem_registry
 
         # We sync to our first registry - if not filesystem, no go
-        if not local.is_filesystem_registry:
+        if not isinstance(local, Filesystem):
```

Comment:
This could also still be None, so maybe
```suggestion
        if not local or not isinstance(local, Filesystem):
```